rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotClientAPI.py

```diff
@@ -0,0 +1,190 @@
+            response.raise_for_status()
+            if self.debug:
+                print(f'Response: {response.json()}')
+            return response.json()['success']
+        except HTTPError as http_err:
+            print(f'HTTP error: {http_err}')
+        except Exception as err:
+            print(f'Other error: {err}')
+        return False
+
+    def start_process(self, process: str, map_name: str):
+        ''' Request the robot to begin a process. This is specific to the robot
+            and the use case. For example, load/unload a cart for Deliverybot
+            or begin cleaning a zone for a cleaning robot.
+            Return True if the robot has accepted the request, else False'''
+        url = self.prefix +\
+            f"/open-rmf/rmf_demos_fm/start_task?robot_name={self.robot_name}"
+        # data fields: task, map_name, destination{}, data{}
+        data = {'task': process, 'map_name': map_name}
+        try:
+            response = requests.post(url, timeout=self.timeout, json=data)
+            response.raise_for_status()
+            if self.debug:
+                print(f'Response: {response.json()}')
+            return response.json()['success']
+        except HTTPError as http_err:
+            print(f'HTTP error: {http_err}')
+        except Exception as err:
+            print(f'Other error: {err}')
+        return False
+
+    def stop(self):
+        ''' Command the robot to stop.
+            Return True if robot has successfully stopped. Else False'''
+        url = self.prefix +\
+            f'/open-rmf/rmf_demos_fm/stop_robot?robot_name={self.robot_name}'
+        try:
+            response = requests.get(url, self.timeout)
+            response.raise_for_status()
+            if self.debug:
+                print(f'Response: {response.json()}')
+            return response.json()['success']
+        except HTTPError as http_err:
+            print(f'HTTP error: {http_err}')
+        except Exception as err:
+            print(f'Other error: {err}')
+        return False
+
+    def navigation_remaining_duration(self):
+        ''' Return the number of seconds remaining for the robot to reach its
+            destination'''
+        response = self.data()
+        if response is not None:
+            return response['data']['destination_arrival_duration']
+        else:
+            return 0.0
+
+    def navigation_completed(self):
+        ''' Return True if the robot has successfully completed its previous
+            navigation request. Else False.'''
+        response = self.data()
+        if response is not None and response.get('data') is not None:
+            return response['data']['completed_request']
+        else:
+            return False
+
+    def process_completed(self):
+        ''' Return True if the robot has successfully completed its previous
+            process request. Else False.'''
+        response = self.data()
```

Comment:
Instead of duplicating the code here, lets just call `return self.navigation_completed()`

Reply:
Done